packages/termux-tools/pkg

```diff
@@ -186,32 +192,56 @@ if [ $# = 0 ]; then
 fi
 
 CMD="$1"
+WRITE_ERROR=false
 shift 1
 
-case "$CMD" in
-	f*) dpkg -L "$@";;
-	h*) show_help;;
-	sh*|inf*) apt show "$@";;
-	add|i*)
-		select_mirror
-		update_apt_cache
-		apt install "$@"
-		;;
-	autoc*) apt autoclean;;
-	cl*) apt clean;;
-	list-a*) apt list "$@";;
-	list-i*) apt list --installed "$@";;
-	rei*) apt install --reinstall "$@";;
-	se*)
-		select_mirror
-		update_apt_cache
-		apt search "$@"
-		;;
-	un*|rem*|rm|del*) apt remove "$@";;
-	up*)
-		select_mirror
-		apt update
-		apt full-upgrade "$@"
-		;;
-	*) echo "Unknown command: '$CMD' (run 'pkg help' for usage information)"; exit 1;;
-esac
+if [ "$TERMUX_MAIN_PACKAGE_FORMAT" = "debian" ]; then
+	case "$CMD" in
+		f*) dpkg -L "$@";;
+		h*) show_help;;
+		sh*|inf*) apt show "$@";;
+		add|i*)
+			select_mirror
+			update_apt_cache
+			apt install "$@"
+			;;
+		autoc*) apt autoclean;;
+		cl*) apt clean;;
+		list-a*) apt list "$@";;
+		list-i*) apt list --installed "$@";;
+		rei*) apt install --reinstall "$@";;
+		se*)
+			select_mirror
+			update_apt_cache
+			apt search "$@"
+			;;
+		un*|rem*|rm|del*) apt remove "$@";;
+		up*)
+			select_mirror
+			apt update
+			apt full-upgrade "$@"
+			;;
+		*) WRITE_ERROR=true;;
+	esac
+else
+	case "$CMD" in
```

Comment:
I would prefer if package manager specific work is carried out in something like `install_package` which handles both apt and pacman specific logic. This will also prevent code duplication as here

Reply:
Is the `install_package` function only responsible for package installation functions?  I just can't understand the meaning of the function and I don't want to turn the code into shit code.

PS: please describe your message in more detail, I just don't understand some of them